test/framework/src/main/java/org/elasticsearch/index/engine/TranslogHandler.java

```diff
@@ -117,20 +118,22 @@ public int run(Engine engine, Translog.Snapshot snapshot) throws IOException {
         return opsRecovered;
     }
 
-    private Engine.Operation convertToEngineOp(Translog.Operation operation, Engine.Operation.Origin origin) {
+    public Engine.Operation convertToEngineOp(Translog.Operation operation, Engine.Operation.Origin origin) {
         switch (operation.opType()) {
             case INDEX:
                 final Translog.Index index = (Translog.Index) operation;
                 final String indexName = mapperService.index().getName();
                 final Engine.Index engineIndex = IndexShard.prepareIndex(docMapper(index.type()),
                     new SourceToParse(indexName, index.type(), index.id(), index.source(), XContentHelper.xContentType(index.source()),
                             index.routing()), index.seqNo(), index.primaryTerm(),
-                        index.version(), null, origin, index.getAutoGeneratedIdTimestamp(), true, SequenceNumbers.UNASSIGNED_SEQ_NO, 0);
+                        index.version(), origin == Engine.Operation.Origin.PRIMARY ? VersionType.EXTERNAL : null,
```

Comment:
Perhaps write the code the same way as in IndexShard:

```
// If a translog op is replayed on the primary (eg. ccr), we need to use external instead of null for its version type.
final VersionType versionType = (origin == Engine.Operation.Origin.PRIMARY) ? VersionType.EXTERNAL : null;
```

Reply:
++. Fixed in https://github.com/elastic/elasticsearch/pull/47188/commits/f91116c2ec656cbd30e511dc28b7f3f7bb248da1.